docs_src/design/adr/devops/0007-Release-Automation.md

```diff
@@ -0,0 +1,87 @@
+
+In Review for Geneva
+
+## Context
+
+EdgeX Foundry is a framework of microservices. As project our project grows, the number of artifacts to release increases as well. This proposal outlines the automation of the most common artifacts that we release.
+
+## Requirements
+
+### Release Artifact Definition
+
+For the scope of Geneva release artifact types are defined as:
+
+- GitHub tags in the repositories in Github.
+- Docker images in our Nexus repository and Docker hub.
+- Snaps in the Snapcraft store.
+
+This list is likely to expand in future releases.
+
+### General Requirements
+
+As the EdgeX Release Czar I gathered the following requirements for automating this part of the release.
+
+1. The release automation needs a manual trigger to be triggered by the EdgeX Release Czar or the Linux Foundation Release Engineers. The goal of this automation is to have a "push button" release mechanism to reduce human error in our release process.
+2. Release artifacts can come from one or more GitHub repository at a time.
+3. GitHub repositories can have one or more release artifact type to release.
+4. GitHub repositories can have one or more artifacts of a specific type to release. (For example: The mono repository, edgex-go, has more than 20 docker images to release.)
+5. GitHub repositories may be released at different times. (For example: Application and Device service repositories can be released on a different day than the Core services in the mono repository.)
+6. Ability to track multiple release streams for the project.
+7. An audit trail history for releases.
+
+## Location
+
+The code that will manage the release automation for EdgeX Foundry will live in a repository called `cd-management`. This repository will have a branch named `release` that will track the releases of artifacts off the `master` branch of the EdgeX Foundry repositories.
+
+### Multiple Release Streams
+
+EdgeX Foundry has this idea of multple release streams that basically coincides with different named branches in GitHub. For the majority of the main releases we will be targeting those off the `master` branch. In the future we will mark a specific release for long term support (LTS). When this happens we will have to branch off `master` in the EdgeX repositories and create a seperate release stream for the LTS. The suggestion at that point will be to branch off the `release` branch in `cd-management` and use this new release branch to track the LTS branches in the EdgeX repositories.
+
+## Release Flow
+
+### Go Modules, Device and Application SDKs
+
+#### During Development
+
+![Merge Actions](0007/gomods_mergeactions.png)
+
+Go modules, Application and Device SDKs only release a GitHub tag as their release. Go modules are set up to automatically release a new patch version on every merge into the `master` branch of the repository. (IE: 1.0.0 -> 1.0.1) For Application and Device SDKs we increment a developmental version tag. (IE: 1.0.0-dev.1 -> 1.0.0-dev.2)
```

Comment:
No the go modules are not released and versioned with edgex-go. Each module has its own separate cadence. The modules are released and versioned more aggressively because they are used in all the services in EdgeX. Tying them to the core services cadence would be detrimental to development because a go module goes through revisions much faster than the microservices it supports.